pkg/azurefile/controllerserver.go

```diff
@@ -491,27 +494,33 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 		mc.ObserveOperationWithResult(isOperationSucceeded, VolumeID, volumeID)
 	}()
 
-	klog.V(2).Infof("begin to create file share(%s) on account(%s) type(%s) subID(%s) rg(%s) location(%s) size(%d) protocol(%s)", validFileShareName, accountName, sku, subsID, resourceGroup, location, fileShareSize, shareProtocol)
-	if err := d.CreateFileShare(ctx, accountOptions, shareOptions, secret); err != nil {
-		if strings.Contains(err.Error(), accountLimitExceedManagementAPI) || strings.Contains(err.Error(), accountLimitExceedDataPlaneAPI) {
-			klog.Warningf("create file share(%s) on account(%s) type(%s) subID(%s) rg(%s) location(%s) size(%d), error: %v, skip matching current account", validFileShareName, accountName, sku, subsID, resourceGroup, location, fileShareSize, err)
-			tags := map[string]*string{
-				azure.SkipMatchingTag: pointer.String(""),
-			}
-			if rerr := d.cloud.AddStorageAccountTags(ctx, subsID, resourceGroup, accountName, tags); rerr != nil {
-				klog.Warningf("AddStorageAccountTags(%v) on account(%s) subsID(%s) rg(%s) failed with error: %v", tags, accountName, subsID, resourceGroup, rerr.Error())
-			}
-			// release volume lock first to prevent deadlock
-			d.volumeLocks.Release(volName)
-			// clean search cache
-			if err := d.accountSearchCache.Delete(lockKey); err != nil {
-				return nil, status.Errorf(codes.Internal, err.Error())
+	if req.GetVolumeContentSource() != nil {
+		if err := d.copyVolume(ctx, req, accountKey, shareOptions, storageEndpointSuffix, secret); err != nil {
```

Comment:
I may be blind, where is the destination share actually created? And how do you make sure it's deleted when `azcopy` permanently fails and you want to return a final error to the external-provisioner? Because the external-provisioner may not call CreateVolume again if user has deleted their PVC, as the volume is not needed any longer.

Reply:
yeah, but where is the share _deleted_ after failed `CopyFileShare`?

Reply:
the new file share would be created by `azcopy`, and if azcopy returns failure, it does not makes sense to delete  the file share explicitly by the csi driver since the csi driver would retry volume cloning after the failure within a period of time.

Reply:
That means that the driver leaks a volume when user deletes corresponding pvc. I understand the interface between the provisioner and the driver does not allow much options here, perhaps we can enhance the protocol if you have a smart idea.